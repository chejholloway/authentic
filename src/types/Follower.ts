/**
 * Represents a GitHub follower's information.
 *
 * This interface defines the structure of information associated with a GitHub follower.
 * It includes details such as the follower's username, profile URL, avatar URL, and more.
 *
 * @interface Follower
 * @example
 * ```ts
 * const follower: Follower = {
 *   login: "exampleFollower",
 *   id: 456,
 *   avatar_url: "https://example.com/follower-avatar.jpg",
 *   // ...other properties
 * };
 * ```
 */
export default interface Follower {
  login: string;
  id: number;
  node_id: string;
  avatar_url: string;
  gravatar_id: string;
  url: string;
  html_url: string;
  followers_url: string;
  following_url: string;
  gists_url: string;
  starred_url: string;
  subscriptions_url: string;
  organizations_url: string;
  repos_url: string;
  events_url: string;
  received_events_url: string;
  type: string;
  site_admin: boolean;
}
