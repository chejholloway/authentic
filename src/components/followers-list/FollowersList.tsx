import React from "react";
import { Link } from "react-router-dom";
import { Avatar } from "@material-tailwind/react";

import Follower from "../../types/Follower";
import { GrayText, Title } from "../../styles/TwinStyles.styles";

interface FollowersListProps {
  followers: Follower[] | undefined;
}

const FollowersList: React.FC<FollowersListProps> = ({ followers }) => {
  return (
    <GrayText>
      <Title className="font-normal">Followers</Title>
      <div className="flex flex-wrap space-x-4">
        {followers ? (
          followers.map((follower) => (
            <div key={follower.id} className="m-4">
              <Link to={follower?.html_url}>
                <Avatar
                  variant="circular"
                  title={`${follower?.login}'s Avatar`}
                  alt={follower?.login}
                  src={follower?.avatar_url}
                />
              </Link>
            </div>
          ))
        ) : (
          <p>No followers available.</p>
        )}
      </div>
    </GrayText>
  );
};

export default FollowersList;
