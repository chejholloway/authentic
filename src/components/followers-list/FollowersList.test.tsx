import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Avatar, Typography } from '@material-tailwind/react';
import FollowersList from './FollowersList';
import Follower from '../../types/Follower';

jest.mock('@material-tailwind/react', () => ({
  Avatar: ({ alt }: { alt: string }) => <img alt={alt} />,
  Typography: ({ children }: { children: React.ReactNode }) => <p>{children}</p>,
  Link: ({ children }: { children: React.ReactNode }) => <a>{children}</a>,
}));

describe('<FollowersList>', () => {
  test('renders followers list with data', () => {
    const mockFollowers: Follower[] = [
    {
      "login": "follower1",
      "id": 851,
      "node_id": "MDQ6VXNlcjg1MQ==",
      "avatar_url": "https://avatars.githubusercontent.com/u/851?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/mountain",
      "html_url": "https://github.com/mountain",
      "followers_url": "https://api.github.com/users/mountain/followers",
      "following_url": "https://api.github.com/users/mountain/following{/other_user}",
      "gists_url": "https://api.github.com/users/mountain/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/mountain/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/mountain/subscriptions",
      "organizations_url": "https://api.github.com/users/mountain/orgs",
      "repos_url": "https://api.github.com/users/mountain/repos",
      "events_url": "https://api.github.com/users/mountain/events{/privacy}",
      "received_events_url": "https://api.github.com/users/mountain/received_events",
      "type": "User",
      "site_admin": false
    },
    {
      "login": "follower2",
      "id": 1014,
      "node_id": "MDQ6VXNlcjEwMTQ=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1014?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/sunfmin",
      "html_url": "https://github.com/sunfmin",
      "followers_url": "https://api.github.com/users/sunfmin/followers",
      "following_url": "https://api.github.com/users/sunfmin/following{/other_user}",
      "gists_url": "https://api.github.com/users/sunfmin/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/sunfmin/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/sunfmin/subscriptions",
      "organizations_url": "https://api.github.com/users/sunfmin/orgs",
      "repos_url": "https://api.github.com/users/sunfmin/repos",
      "events_url": "https://api.github.com/users/sunfmin/events{/privacy}",
      "received_events_url": "https://api.github.com/users/sunfmin/received_events",
      "type": "User",
      "site_admin": false
    },
  ];

    const { asFragment } = render(
      <Router>
        <FollowersList followers={mockFollowers} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders "No followers available." when followers are undefined', () => {
    const { asFragment } = render(
      <Router>
        <FollowersList followers={undefined} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
