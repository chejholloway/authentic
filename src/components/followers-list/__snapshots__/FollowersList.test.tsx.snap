// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowersList> renders "No followers available." when followers are undefined 1`] = `
<DocumentFragment>
  <div
    class="text-gray-700"
  >
    <h3
      class="text-center"
    >
      <p>
        Followers
      </p>
    </h3>
    <div
      class="flex flex-wrap space-x-4"
    >
      <p>
        No followers available.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<FollowersList> renders followers list with data 1`] = `
<DocumentFragment>
  <div
    class="text-gray-700"
  >
    <h3
      class="text-center"
    >
      <p>
        Followers
      </p>
    </h3>
    <div
      class="flex flex-wrap space-x-4"
    >
      <div>
        <a
          href="/"
        >
          <img
            alt="follower1"
          />
        </a>
      </div>
      <div>
        <a
          href="/"
        >
          <img
            alt="follower2"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
